src/core/resources/assets/userAssets.ts

```
@@ -126,6 +165,66 @@ async function userAssetsQueryFunction({
       testnetMode,
     }),
   })?.state?.data || {}) as ParsedAssetsDictByChain;
+
+  // For test wallets in testing mode, immediately return stubbed assets
+  // This ensures tests have predictable balances even if anvil balances are slow or fail
+  if (process.env.IS_TESTING === 'true' && isTestWallet(address)) {
```

Worried about this approach because it essentially changes the code path of testing vs prod, so prod issues could still silently fail because of a conditional here. I think the simplest path forward would be to adopt `mockFetch()` from [this PR](https://github.com/rainbow-me/browser-extension/pull/1754/files#diff-894d4380731ad45d7511aa641d4b973f15f8498294ca1c8ef11f4e66e2ba001b) similarly to how we mock swaps. mockFetch is already registered for all `IS_TESTING` runs, so all we would need to do is register the unique userAsset API request URLs (unique for each wallet address already), generate some mocks, and expose them to `mockFetch()`. We could even do a little refactoring to standardize this so that `fetchResponses` and `mockFetch` are fully agnostic and we can register more service mocks